jquery.ajaxchimp.js

```diff
@@ -93,15 +94,31 @@ For e.g. 'http://blahblah.us1.list-manage.com/subscribe/post-json?u=5afsdhfuhdsi
                         }
                     }
 
+
                     // Translate and display message
+                    var length = 0, msgnr;
+
+                    if(Object.keys){
+                        length = Object.keys($.ajaxChimp.responses).length
+                    }
+                    else {
+                        for (var key in $.ajaxChimp.responses){
+                            if ($.ajaxChimp.responses.hasOwnProperty(key)) {
+                              length++;
+                            }
+                        }
+                    }
+
+                    while(length--) {
+                        if (msg.indexOf($.ajaxChimp.responses[length])!==-1) {
+                            msgnr = length;
+                        }
+                    }
```

Comment:
Also, I think the fix needs to be a lot better. Meaning, there are cases where the response is like:
```json
{"result":"error","msg":"test@example.com is already subscribed to list My Newsletter. <a href=\"http:\/\/address.list-manage1.com\/subscribe\/send-email?u=uuu&id=ddd&e=xxx==\">Click here to update your profile.<\/a>"})
```

I believe the best fix would be not to do partial matches (like I did also) but rather regular expressions to extract the information we need.

BTW, any luck on requesting Mailchimp to return these errors already translated from the server? Basically the translations exist and can be tweaked from the Mailchimp webpage. I still don't understand why they don't translate this before sending the response. Perhaps we could send a language param to get the response from them directly?

The reason why I'm saying this is because I might have customers that want to translate the same errors in different ways and that way I wouldn't have to change the translations per customer, and rather just tell them how to do on Mailchimp interface.